lucene/codecs/src/java/org/apache/lucene/codecs/customcompression/ZstdDictCompressionMode.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.codecs.customcompression;
+
+import com.github.luben.zstd.*;
+import java.io.IOException;
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.compressing.Compressor;
+import org.apache.lucene.codecs.compressing.Decompressor;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.DataOutput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+
+/** Zstandard Compression Mode */
+public class ZstdDictCompressionMode extends CompressionMode {
+  private static final int NUM_SUB_BLOCKS = 10;
+  private final int level;
+  public static final int defaultLevel = 3;
+  private static final int DICT_SIZE_FACTOR = 2;
```

Comment:
Let's do 6 like for BEST_COMPRESSION?

```suggestion
  private static final int DICT_SIZE_FACTOR = 6;
```

Reply:
Changed to 6 in PR